server/src/test/java/org/elasticsearch/transport/TransportServiceHandshakeTests.java

```diff
@@ -190,4 +307,59 @@ public void testNodeConnectWithDifferentNodeId() {
         }
     }
 
+    private static Version randomCompatibleVersion() {
+        return VersionUtils.randomVersionBetween(random(), Version.CURRENT.minimumCompatibilityVersion(), Version.CURRENT);
+    }
+
+    private static InboundMessage unreadableHandshake(InboundMessage message) {
+        return new InboundMessage(message.getHeader(), false) {
+            @Override
+            public int getContentLength() {
+                return message.getContentLength();
+            }
+
+            @Override
+            public StreamInput openOrGetStreamInput() {
```

Comment:
I think we can do this way shorter as:

```java
            @Override
            public StreamInput openOrGetStreamInput() {
                final StreamInput streamInput = new InputStreamStreamInput(new InputStream() {
                    @Override
                    public int read() {
                        throw new ElasticsearchException("unreadable handshake");
                    }
                });
                streamInput.setVersion(message.getHeader().getVersion());
                return streamInput;
            }
```

Reply:
Thanks, I was pretty sure there was a way to do this by overriding a single method somewhere.